docs/design/logging-integration.adoc

```diff
@@ -0,0 +1,196 @@
+= Logging Integration
+
+== Overview
+
+KubeArchive needs to support logging, but it is not a logging system itself and
+does not implement logging. Instead, KubeArchive will integrate with logging systems
+and provide URLs for retrieving log files from the logging system for a specific
+Kubernetes resource.
+
+It is important to note that logs are tied to `Pods`. When a user requests the logs
+for a Tekton `PipelineRun`, what they expect to get back are the logs attached to the
+`Pods` that were part of the `PipelineRun`. Similar cases exist for requesting logs for
+`Jobs` and `CronJobs`. KubeArchive has to be able to handle this seamlessly for the user.
+
+== Retrieving Log Information
+
+In generic terms, this can be done using the owner reference field in a resource, a
+sort of backwards recursive search. When a `PipelineRun` is deleted, all the `TaskRuns`
+associated with the `PipelineRun`, and `Pods` associated with those `TaskRuns`, are
+deleted. This is done using the owner references.
+
+KubeArchive can do things similarly. When logs for a resource are requested, a query
+is made to find all the resources that have that initial resource as an owner.  Then
+each resource returned is processed similarly, eventually building up a list of log
+file links.  This generic approach should work for any resource.
+
+This also implies that KubeArchive is configured correctly to support this. It must
+be configured so that the initial resource and any dependent resources, all the way
+down to and including the `Pods`, are archived.
+
+Here's a sample `KubeArchiveConfig` as an example:
+```
+---
+apiVersion: kubearchive.kubearchive.org/v1alpha1
+kind: KubeArchiveConfig
+metadata:
+  name: kubearchive
+  namespace: test
+spec:
+  resources:
+    - deleteWhen: has(status.completionTime)
+      selector:
+        apiVersion: ""
+        kind: CronJob
+    - archiveOnDelete: true
+      selector:
+        apiVersion: ""
+        kind: Pod
+```
```

Comment:
In Asciidoc [code blocks](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/) should be written like this:
```suggestion
[source,yaml]
----
---
apiVersion: kubearchive.kubearchive.org/v1alpha1
kind: KubeArchiveConfig
metadata:
  name: kubearchive
  namespace: test
spec:
  resources:
    - deleteWhen: has(status.completionTime)
      selector:
        apiVersion: ""
        kind: CronJob
    - archiveOnDelete: true
      selector:
        apiVersion: ""
        kind: Pod
----
```

Reply:
Fixed.